screenpy/resolutions/matches.py

```diff
@@ -17,14 +24,21 @@ class Matches:
         )
     """
 
+    @property
+    def item_to_log(self) -> str:
+        """Represent the item in a log-friendly way."""
+        if isinstance(self.pattern, Pattern):
+            return f"r'{self.pattern.pattern}'"
+        return f"r'{self.pattern}'"
```

Comment:
Do we want to move this logic to `represent_prop`?

Reply:
There is a slight issue with doing that.  I thought it best to show the pattern in the log as `r'pattern'`, which includes the `r` before the string quotes.  This is doable with `Pattern` objects but  when the pattern is a string, there is no way for `represent_prop` to differentiate r-strings from regular strings.

```python
    @property
    def pattern_to_log(self) -> str:
        """Represent the item in a log-friendly way."""
        return represent_prop(self.pattern)
```
The above will fail the logging tests:
```
Expected :"Text matching the pattern r'(spam)+'."
Actual   :"Text matching the pattern '(spam)+'."
```

To counter this, I considered adding a switch argument to `represent_prop` which would gives the caller the option to turn on/off the `r` in front of strings but I felt like this was an over complicated solution for a single prop.

Reply:
When `self.pattern` is a string, `represent_prop` will call `repr`
```python
    if isinstance(item, str):
        return repr(item)
```
Which is unfortunately not how we want the raw string to appear in the log.

`represent(r"\tpattern")` would return `"\\tpattern"`.  Putting an `r` in front of the string wouldn't fix that.